client.go

```diff
@@ -141,7 +141,7 @@ func (c *client) listen() {
 }
 
 func (c *client) handlePacket(remoteAddr net.Addr, packet []byte) error {
-	if protocol.ByteCount(len(packet)) > protocol.MaxPacketSize {
+	if protocol.ByteCount(len(packet)) > protocol.MaxReceivePacketSize {
```

Comment:
Can this actually ever happen if we only read packets of a certain size? Can you look into whether there's some kind of signal from the kernel when packets got truncated?